src/plugins/explore/public/components/visualizations/utils/utils.ts

```diff
@@ -269,5 +269,186 @@ export const getTooltipFormat = (
   fallback = '%b %d, %Y %H:%M:%S'
 ): string => {
   const timeUnit = inferTimeUnitFromTimestamps(data, field);
-  return timeUnit ? timeUnitToFormat[timeUnit] ?? fallback : fallback;
+  return timeUnit ? timeUnitToTooltipFormat[timeUnit] ?? fallback : fallback;
+};
+
+/**
+ * Determines the format for a temporal axis based on the inferred time unit.
+ * @param data The transformed data array
+ * @param field The field name for the temporal axis
+ * @param fallback The fallback format string (default: '%d')
+ * @returns The format string for the axis
+ */
+export const getAxisTemporalFormat = (
+  data: Array<Record<string, any>>,
+  field: string | undefined,
+  fallback = '%H:%M'
+): string => {
+  const timeUnit = inferAxisTimeUnitFromRange(data, field);
+  return timeUnit ? timeUnitToAxisFormat[timeUnit] ?? fallback : fallback;
+};
+
+type TimeUnitName = 'second' | 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year';
+
+export const DURATION = {
+  second: 1000,
+  minute: 1000 * 60,
+  hour: 1000 * 60 * 60,
+  day: 1000 * 60 * 60 * 24,
+  week: 1000 * 60 * 60 * 24 * 7,
+  month: 1000 * 60 * 60 * 24 * 30,
+  year: 1000 * 60 * 60 * 24 * 365,
+} as const;
+
+const TICK_INTERVALS: Array<{ unit: TimeUnitName; step: number; durMs: number }> = [
+  { unit: 'second', step: 1, durMs: 1 * DURATION.second },
+  { unit: 'second', step: 5, durMs: 5 * DURATION.second },
+  { unit: 'second', step: 10, durMs: 10 * DURATION.second },
+  { unit: 'second', step: 15, durMs: 15 * DURATION.second },
+  { unit: 'second', step: 30, durMs: 30 * DURATION.second },
+
+  { unit: 'minute', step: 1, durMs: 1 * DURATION.minute },
+  { unit: 'minute', step: 5, durMs: 5 * DURATION.minute },
+  { unit: 'minute', step: 15, durMs: 15 * DURATION.minute },
+  { unit: 'minute', step: 30, durMs: 30 * DURATION.minute },
+
+  { unit: 'hour', step: 1, durMs: 1 * DURATION.hour },
+  { unit: 'hour', step: 3, durMs: 3 * DURATION.hour },
+  { unit: 'hour', step: 6, durMs: 6 * DURATION.hour },
+  { unit: 'hour', step: 12, durMs: 12 * DURATION.hour },
+
+  { unit: 'day', step: 1, durMs: 1 * DURATION.day },
+  { unit: 'day', step: 2, durMs: 2 * DURATION.day },
+
+  { unit: 'week', step: 1, durMs: 1 * DURATION.week },
+
+  { unit: 'month', step: 1, durMs: 1 * DURATION.month },
+  { unit: 'month', step: 3, durMs: 3 * DURATION.month },
+
+  { unit: 'year', step: 1, durMs: 1 * DURATION.year },
+];
+
+export function pickTickCount(rangeMs: number, targetMin = 10, targetMax = 20): number {
```

Comment:
Can simply use ticket count values like: "millisecond", "second", "minute", "hour", "day", "week", "month", and "year"?